test/unit/utils/sql.test.js

```diff
@@ -109,14 +139,83 @@ describe('injectReplacements (named replacements)', () => {
     });
   });
 
-  it('does not consider the token to be a replacement if it is part of a string with a backslash escaped quote', () => {
-    const sql = injectReplacements('SELECT * FROM users WHERE id = \'\\\':id\' OR id = :id', dialect, {
-      id: 1
-    });
-
-    expectsql(sql, {
-      default: 'SELECT * FROM users WHERE id = \'\\\':id\' OR id = 1'
-    });
+  it('does not consider the token to be a bind parameter if it is part of a string with a backslash escaped quote, in dialects that support backslash escape', () => {
+    expectPerDialect(
+      () =>
+        injectReplacements(
+          "SELECT * FROM users WHERE id = '\\' $id' OR id = $id",
+          dialect,
+          { id: 1 }
+        ),
+      {
+        default:
+          new Error(`The following SQL query includes an unterminated string literal:
+SELECT * FROM users WHERE id = '\\' $id' OR id = $id`),
+        'mysql mariadb': toMatchSql(
+          "SELECT * FROM users WHERE id = '\\' $id' OR id = $id"
+        )
+      }
+    );
+  });
+
+  it('does not consider the token to be a bind parameter if it is part of a string with a backslash escaped quote, in dialects that support standardConformingStrings = false', () => {
+    expectPerDialect(
+      () =>
+        injectReplacements(
+          "SELECT * FROM users WHERE id = '\\' $id' OR id = $id",
+          getNonStandardConfirmingStringDialect(),
+          { id: 1 }
+        ),
+      {
+        default:
+          new Error(`The following SQL query includes an unterminated string literal:
+SELECT * FROM users WHERE id = '\\' $id' OR id = $id`),
+
+        'mysql mariadb postgres': toMatchSql(
+          "SELECT * FROM users WHERE id = '\\' $id' OR id = $id"
+        )
+      }
+    );
+  });
+
+  it('does not consider the token to be a bind parameter if it is part of an E-prefixed string with a backslash escaped quote, in dialects that support E-prefixed strings', () => {
+    expectPerDialect(
+      () =>
+        injectReplacements(
+          "SELECT * FROM users WHERE id = E'\\' $id' OR id = $id",
+          dialect,
+          { id: 1 }
+        ),
+      {
+        default:
+          new Error(`The following SQL query includes an unterminated string literal:
+SELECT * FROM users WHERE id = E'\\' $id' OR id = $id`),
+        'mysql mariadb': toMatchSql(
+          "SELECT * FROM users WHERE id = E'\\' $id' OR id = $id"
+        ),
+        postgres: "SELECT * FROM users WHERE id = E'\\' $id' OR id = $id"
+      }
+    );
+  });
```

Comment:
What is the difference between mysql/mariadb and postgres?

Reply:
probably none :) there was a diff in the original version but that was due to a difference in functionality where the function would return bind values. let me adjust this case